network/builder/src/builder.rs

```diff
@@ -492,13 +490,19 @@ impl NetworkBuilder {
     /// can be attached to other components.
     pub fn add_protocol_handler<SenderT, EventT>(
         &mut self,
-        (rpc_protocols, direct_send_protocols, queue_preference, max_queue_size_per_peer, counter): (
-            Vec<ProtocolId>,
-            Vec<ProtocolId>,
-            QueueStyle,
-            usize,
-            Option<&'static IntCounterVec>,
-        ),
+        (
+                rpc_protocols,
+                direct_send_protocols,
+                queue_preference,
+                max_queue_size_per_peer,
+                counter,
+            ): (
+                Vec<ProtocolId>,
+                Vec<ProtocolId>,
+                QueueStyle,
+                usize,
+                Option<&'static IntCounterVec>,
+            ),
```

Comment:
probably best to make a type here?

Reply:
yes.  Probably, maybe.   That can be done in an orthgonal PR.

Reply:
This would be good as a type and the function itself would be very good as a `pub trait NetworkClientService` or something along those lines;  a significant difficulty on that front is `onchain_discovery` (and `simple_onchain_discovery`) being separate crates within networking.

This specific piece would be significantly simpler/cleaner if we moved `onchain_discovery` properly into the networking crate (at which point we could move `network_builder` back into the networking crate as well)